assets/scripts/build_gallery.py

```diff
@@ -163,6 +214,148 @@ def get_expectation_file_info_dict(
+                continue
+
+        expectation_instances[
+            expectation
+        ] = great_expectations.expectations.registry.get_expectation_impl(expectation)()
+    return expectation_instances
+
+
+def combine_backend_results(
+    expectations_info, expectation_instances, diagnostic_objects, outfile_name
+):
+    expected_full_backend_files = [
+        f"{backend}_full.json" for backend in ALL_GALLERY_BACKENDS
+    ]
+    found_full_backend_files = glob("*_full.json")
+
+    if sorted(found_full_backend_files) == sorted(expected_full_backend_files):
+        logger.info(
+            f"All expected *_full.json files were found. Going to combine and write to {outfile_name}"
+        )
+
+        bad_key_names = []
+        for fname in found_full_backend_files:
+            with open(fname, "r") as fp:
+                text = fp.read()
+            data = json.loads(text)
+
+            for expectation_name in data:
+                try:
+                    expectations_info[expectation_name][
+                        "backend_test_result_counts"
+                    ].extend(data[expectation_name]["backend_test_result_counts"])
+                except KeyError:
+                    expectations_info[expectation_name][
+                        "backend_test_result_counts"
+                    ] = data[expectation_name]["backend_test_result_counts"]
+
+        # Re-calculate maturity_checklist, library_metadata, and coverage_score
+        for expectation_name in expectations_info:
+            try:
+                diagnostic_object = diagnostic_objects[expectation_name]
+            except KeyError:
+                logger.error(f"No diagnostic obj for {expectation_name}")
+                bad_key_names.append(expectation_name)
+                continue
+            expectation_instance = expectation_instances[expectation_name]
+            try:
+                backend_test_result_counts_object = [
+                    ExpectationBackendTestResultCounts(**backend_results)
+                    for backend_results in expectations_info[expectation_name][
+                        "backend_test_result_counts"
+                    ]
+                ]
+            except KeyError:
+                logger.error(f"No backend_test_result_counts for {expectation_name}")
+                bad_key_names.append(expectation_name)
+                continue
+            maturity_checklist_object = Expectation._get_maturity_checklist(
+                expectation_instance=expectation_instance,
+                library_metadata=diagnostic_object.library_metadata,
+                description=diagnostic_object.description,
+                examples=diagnostic_object.examples,
+                tests=diagnostic_object.tests,
+                backend_test_result_counts=backend_test_result_counts_object,
+                execution_engines=diagnostic_object.execution_engines,
+            )
```

Comment:
Could we use the instance to get the maturity checklist since we already have access to it? Then we can eliminate some of the changes in `expectation.py`:
```suggestion
            maturity_checklist_object = expectation_instance._get_maturity_checklist(
                library_metadata=diagnostic_object.library_metadata,
                description=diagnostic_object.description,
                examples=diagnostic_object.examples,
                tests=diagnostic_object.tests,
                backend_test_result_counts=backend_test_result_counts_object,
                execution_engines=diagnostic_object.execution_engines,
            )
```